modules/vpc-sc/main.tf

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  access_policy_name = try(google_access_context_manager_access_policy.default[var.access_policy_title].name, null)
+
+  standard_perimeters = {
+    for key, value in var.perimeters :
+    key => value if value.type == "PERIMETER_TYPE_REGULAR"
+  }
+
+  bridge_perimeters = {
+    for key, value in var.perimeters :
+    key => value if value.type == "PERIMETER_TYPE_BRIDGE"
+  }
+
+  perimeter_access_levels_enforced = try(transpose(var.access_level_perimeters.enforced), null)
+  perimeter_access_levels_dry_run  = try(transpose(var.access_level_perimeters.dry_run), null)
+}
+
+resource "google_access_context_manager_access_policy" "default" {
+  for_each  = toset([var.access_policy_title])
```

Comment:
why are you using `foreach` on a required variable? It can only ever create a single resource since the variable is needed, or am I missing something?

Reply:
`for_each` removed.